File: TestRestApi/src/main/java/com/amuyu/testrestapi/network/DataManagerImpl.java
package com.amuyu.testrestapi.network;

import java.util.List;
import java.util.Map;

import okhttp3.RequestBody;
import okhttp3.ResponseBody;
import retrofit2.Response;
import rx.Observable;

/**
 * Created by noco on 2016-10-27.
 */
public class DataManagerImpl implements DataManager {

    private final ApiManager apiManager;

    public DataManagerImpl() {
        this.apiManager = new ApiManager();
    }


    @Override
    public Observable<Response<ResponseBody>> callApi(String url, Map<String, String> parameter, Map<String,String> headerMap) {
        return apiManager.callApi(url,parameter,headerMap);
    }

    @Override
    public Observable<Response<ResponseBody>> callApiPost(String url, RequestBody body, Map<String,String> headerMap) {
        return apiManager.callApiPost(url,body, headerMap);
    }

    @Override
    public List<String> loadList() {
        return null;
    }
}

File: app/src/main/kotlin/com/lazycouple/restapiclient/repository/api/DataManagerImpl.kt
package com.amuyu.testrestapi.network

import okhttp3.RequestBody
import okhttp3.ResponseBody
import retrofit2.Response
import rx.Observable

/**
 * Created by noco on 2016-10-27.
 */
class DataManagerImpl : DataManager {

    private val apiManager: ApiManager

    init {
        this.apiManager = ApiManager()
    }


    override fun callApi(url: String, parameter: Map<String, String>, headerMap: Map<String, String>): Observable<Response<ResponseBody>>? {
        return apiManager?.callApi(url, parameter, headerMap)
    }

    override fun callApiPost(url: String, body: RequestBody, headerMap: Map<String, String>): Observable<Response<ResponseBody>>? {
        return apiManager?.callApiPost(url, body, headerMap)
    }

    override fun loadList(): List<String>? {
        return null
    }
}
